docs/DeveloperGuide.md

```diff
@@ -1,38 +1,180 @@
+3. **Nuscents**: The main application class that receives and executes commands. It invokes the `execute()` method of the `HelpCommand`.
+4. **UI**: Manages user interface interactions, such as displaying the help menu.
+
+#### II. Component-Level Design
+1. **Parser**:  
+   The `Parser` class recognizes the user's intention to access the help menu through the `help` keyword.
+
+2. **HelpCommand**:  
+   When the `Parser` identifies a `help` command, it instantiates a `HelpCommand` object. This object encapsulates the user's request to view the command instructions.
+
+3. **Nuscents**:  
+   Upon receiving the `HelpCommand` object, the `Nuscents` class triggers the `execute()` method of the `HelpCommand`.
+
+4. **UI**:  
+   The `UI` class is then responsible for fetching the `HELP_MENU` static string from the `HelpCommand` class and displaying it to the user. This ensures the user receives a comprehensive list of commands available in the application.
+
+#### III. Alternatives Considered
+Initially, we pondered over embedding the help details directly within the main application class, `Nuscents`. This would eliminate the need for a separate `HelpCommand` class. However, segregating the `HelpCommand` ensures better modularity, making future expansions or modifications seamless.
+
+### `helpCommand` Usage Scenario
+
+1. **Step 1**: The user launches the application. The initial screen appears.
+2. **Step 2**: Unsure of the commands, the user inputs the `help` command.
+3. **Step 3**: The application recognizes the command through the `Parser` and creates a `HelpCommand` object.
+4. **Step 4**: The `Nuscents` class invokes the `execute()` method of the `HelpCommand`.
+5. **Step 5**: The `UI` fetches the `HELP_MENU` string and displays the comprehensive list of commands to the user.
 
 
 ## Product scope
-### Target user profile
 
-{Describe the target user profile}
+### Target user profile
+SOC students at NUS who are tech-savvy, with limited allowance to monitor their finances and expense management
+while balancing aspects of university-life, academics and social activities. SOC students are known to be familiar with 
+CLI applications, which are often built to be efficient and clutter-free.
 
 ### Value proposition
-
 {Describe the value proposition: what problem does it solve?}
 
 ## User Stories
-
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ...           | I want to ...                                             | So that I can ...                                        |
+|---------|--------------------|-----------------------------------------------------------|----------------------------------------------------------|
```

Comment:
<img width="791" alt="image" src="https://github.com/nus-cs2113-AY2324S1/tp/assets/84222810/eadc8348-b491-4c20-a0aa-e9b76d6e6ef7">

I think your table formatting is broken in the github.io site